packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -528,6 +536,60 @@ export class DefaultCrudRepository<
     return includeRelatedModels<T, Relations>(this, entities, include, options);
   }
 
+  /**
+   * This function works as a persist hook.
+   * It converts an entity from the CRUD operations' caller
+   * to a persistable data that can will be stored in the
+   * back-end database.
+   *
+   * User can extend `DefaultCrudRepository` then override this
+   * function to execute custom persist hook.
+   * @param entity The entity passed from CRUD operations' caller.
+   * @param options
+   */
+  protected async entityToData<R extends T>(
```

Comment:
Since `entityToData` is a new public API, we should have tests to describe and verify the intended behavior:
 - `entityToData` is called for create/patch/replace operations (probably for `delete` too)
 -  the value returned by `entityToData` is sent to persisted model, i.e. modifications made by this function are applied

Thoughts?

Reply:
unit test for `entityToData` is added. It is invoked by `create()` method cause it's protected.